File: src/main/java/it/camera/stampati/model/StampatoModel.java
package it.camera.stampati.model;

import java.util.Date;
import java.util.List;

import it.camera.stampati.enumerator.StampatoFormat;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class StampatoModel {
	
	@NotNull
	private StampatoIdModel id;
	private Long progressivo;
	private Boolean pubblicato;
	private Boolean errataCorrige;
	private String rigoNero;
    private Boolean pdfPresente;
    private Boolean htmlPresente;
    private String numeroAtto;
    private String nomeFrontespizio;
    private String nomeFile;
    private String numeriPDL;
    private Integer pagine;
    private String navette;
    private String lettera;
    private Boolean testoUnificato;
    private Boolean rinvioInCommissione;
    private Boolean relazioneMagg;
    private String relazioneMin;
    private String suffisso;
    private String denominazioneStampato;
    private String tipoPresentazione;
    private Date dataPresentazione;
    private Date presentazioneOrale;
    private Date dataStampa;
    private String titolo;
    private StampatoFormat format;
    private Date dataDeleted;
    private Date createdAt;
    private Date updatedAt;
    private List<StampatoFelModel> stampatiFel;
    private List<StampatoRelatoreModel> stampatiRelatori;
    
    public String getBarcode() {
        return id != null ? id.getBarcode() : null;
    }
}

File: gui/src/app/models/stampato.model.ts
import { Type } from "class-transformer";
import "reflect-metadata";
import { FelRelatoreModel } from "./stampato.fel.model";
import { StampatoRelatoreModel } from "./stampato.relatore.model";

export enum StampatoFormat {
  	PDF = 1,
	XHTML = 2,
	HTML = 3,
	EPUB = 4
}

export class StampatoIdModel {
	barcode: string;
	legislatura: number;
}

export class StampatoModel {
	@Type(() => StampatoIdModel)
	id: StampatoIdModel;
	codiceEstremiAttoPdl?: number;
	progressivo?: number;
	pubblicato?:boolean;
	errataCorrige: boolean;
	rigoNero: string;
	pdfPresente?: boolean;
	htmlPresente?: boolean;
	numeroAtto?: string;
	nomeFrontespizio?: string;
	nomeFile?: string;
	numeriPDL?: string;
	pagine?: number;
	lettera?: string;
	navette?: string;
	testoUnificato?:boolean;
	rinvioInCommissione?: boolean;
	relazioneMagg?: boolean;
	relazioneMin?: string;
	suffisso?: string;
	denominazioneStampato?: string;
	tipoPresentazione?: string;
	@Type(() => Date)
	dataPresentazione?: Date;
	@Type(() => Date)
	presentazioneOrale?: Date;
	@Type(() => Date)
	dataStampa?: Date;
	titolo?: string = '';
	titoloFel?: string = '';
	format?: StampatoFormat;
	@Type(() => Date)
	dataDeleted?: Date;
	@Type(() => Date)
	createdAt?: Date;
	@Type(() => Date)
	updatedAt?: Date;
	@Type(() => FelRelatoreModel)
	relatoriFel?: FelRelatoreModel[];
	@Type(() => StampatoRelatoreModel)
	stampatiRelatori?: StampatoRelatoreModel[];

	constructor(id: StampatoIdModel) {
		this.id = id;
	}
}